samples/net/udp/src/main.c

```diff
@@ -0,0 +1,202 @@
+		"IP address %s, port number %d",
+		CONFIG_UDP_SAMPLE_DATA_UPLOAD_SIZE_BYTES + UDP_IP_HEADER_SIZE,
+		CONFIG_UDP_SAMPLE_SERVER_ADDRESS_STATIC,
+		CONFIG_UDP_SAMPLE_SERVER_PORT);
+
+	err = send(client_fd, buffer, sizeof(buffer), 0);
+	if (err < 0) {
+		LOG_ERR("Failed to transmit UDP packet, %d", errno);
+		return;
+	}
+
+	(void)k_work_reschedule(&server_transmission_work,
+				K_SECONDS(CONFIG_UDP_SAMPLE_DATA_UPLOAD_FREQUENCY_SECONDS));
+}
+
+static int server_init(void)
+{
+	struct sockaddr_in *server4 = ((struct sockaddr_in *)&host_addr);
+
+	server4->sin_family = AF_INET;
+	server4->sin_port = htons(CONFIG_UDP_SAMPLE_SERVER_PORT);
+
+	inet_pton(AF_INET, CONFIG_UDP_SAMPLE_SERVER_ADDRESS_STATIC, &server4->sin_addr);
+
+	return 0;
+}
+
+static void server_disconnect(void)
+{
+	(void)close(client_fd);
+}
+
+static int server_connect(void)
+{
+	int err;
+
+	client_fd = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+	if (client_fd < 0) {
+		LOG_INF("Failed to create UDP socket: %d", errno);
+		err = -errno;
+		goto error;
+	}
+
+	err = connect(client_fd, (struct sockaddr *)&host_addr, sizeof(struct sockaddr_in));
+	if (err < 0) {
+		LOG_INF("connect, error: %d", errno);
+		err = -errno;
+		goto error;
+	}
+
+	return 0;
+
+error:
+	server_disconnect();
+	return err;
+}
+
+static void on_net_event_l4_connected(void)
+{
+	int err;
+
+	err = server_connect();
+	if (err) {
+		LOG_INF("Not able to connect to UDP server");
+		FATAL_ERROR();
+		return;
+	}
+
+	(void)k_work_reschedule(&server_transmission_work, K_NO_WAIT);
+}
+
+static void on_net_event_l4_disconnected(void)
+{
+	/* Disconnect socket */
+	server_disconnect();
+	(void)k_work_cancel_delayable(&server_transmission_work);
+}
+
+static void l4_event_handler(struct net_mgmt_event_callback *cb,
+			     uint32_t event,
+			     struct net_if *iface)
+{
+	switch (event) {
+	case NET_EVENT_L4_CONNECTED:
+		LOG_INF("Network connectivity established");
+		on_net_event_l4_connected();
+		break;
+	case NET_EVENT_L4_DISCONNECTED:
+		LOG_INF("Network connectivity lost");
+		on_net_event_l4_disconnected();
+		break;
+	default:
+		/* Don't care */
+		return;
+	}
+}
+
+static void connectivity_event_handler(struct net_mgmt_event_callback *cb,
+				       uint32_t event,
+				       struct net_if *iface)
+{
+	if (event == NET_EVENT_CONN_IF_FATAL_ERROR) {
+		LOG_ERR("NET_EVENT_CONN_IF_FATAL_ERROR");
+		FATAL_ERROR();
+		return;
+	}
+}
+
+int main(void)
+{
+	int err;
+
+	LOG_INF("UDP sample has started");
+
+	/* Setup handler for Zephyr NET Connection Manager events. */
+	net_mgmt_init_event_callback(&l4_cb, l4_event_handler, L4_EVENT_MASK);
+	net_mgmt_add_event_callback(&l4_cb);
+
+	/* Setup handler for Zephyr NET Connection Manager Connectivity layer. */
+	net_mgmt_init_event_callback(&conn_cb, connectivity_event_handler, CONN_LAYER_EVENT_MASK);
+	net_mgmt_add_event_callback(&conn_cb);
+
+	err = server_init();
+	if (err) {
+		LOG_INF("Not able to initialize UDP server connection");
```

Comment:
The log message is a bit confusing, we're not setting up any connection here, just configuring server address.

Reply:
Agree, I think initially the intention was to keep the nomenclature close to the socket operations that is performed